docs/stateful_models.md

```diff
@@ -0,0 +1,317 @@
+
+
+################   Starting sequence without ID   #################
+
+request.inputs['sequence_control_input'].CopyFrom(
+               make_tensor_proto([SEQUENCE_START], dtype="uint32"))
+
+
+################       Non control requests       #################
+
+request.inputs['sequence_id'].CopyFrom(
+               make_tensor_proto([sequence_id], dtype="uint64"))
+
+
+################         Ending sequence          #################
+
+request.inputs['sequence_control_input'].CopyFrom(
+               make_tensor_proto([SEQUENCE_END], dtype="uint32"))
+request.inputs['sequence_id'].CopyFrom(
+                make_tensor_proto([sequence_id], dtype="uint64"))
+
+###################################################################
+
+# Send request to OVMS and get response
+response = stub.Predict(request, 10.0)
+
+# response variable now contains model outputs (inference results) as well as sequence_id in response.outputs
+
+# Fetch sequence id from the response
+sequence_id = response.outputs['sequence_id'].uint64_val[0]
+
+```
+
+See [grpc_stateful_client.py](../example_client/grpc_stateful_client.py) example client for reference.
+
+### Inference via HTTP <a name="stateful_http"></a>
+
+Inference on stateful models via HTTP is very similar to inference on stateless models (_see [REST API](model_server_rest_api.md#predict-api) for reference_). The difference is that requests to stateful models must containt additional inputs with information necessary for proper sequence handling.
+
+`sequence_id` and `sequence_control_input` must be added to HTTP request by adding new `key:value` pair in `inputs` field of JSON body. 
+
+For both inputs value must be a single number in 1-dimensional array.
+
+Example: (_using Python requests package_):
+
+```
+...
+import json
+import requests
+...
+
+SEQUENCE_START = 1
+SEQUENCE_END = 2
+sequence_id = 10
+
+
+inputs = {}
+
+""" 
+Add inputs with data to infer
+"""
+
+################   Add stateful specific inputs   #################
+
+################ Starting sequence with custom ID #################
+
+inputs['sequence_control_input'] = [int(SEQUENCE_START)]
+inputs['sequence_id'] = [int(sequence_id)]
+
+
+################   Starting sequence without ID   #################
+
+inputs['sequence_control_input'] = [int(SEQUENCE_START)]
+
+
+################       Non control requests       #################
+
+inputs['sequence_id'] = [int(sequence_id)]
+
+
+################         Ending sequence          #################
+
+inputs['sequence_control_input'] = [int(SEQUENCE_END)]
+inputs['sequence_id'] = [int(sequence_id)]
+
+###################################################################
+
+# Prepare request
+signature = "serving_default"
+request_body = json.dumps({"signature_name": signature,'inputs': inputs})
+
+# Send request to OVMS and get response
+response = requests.post("localhost:5555/v1/models/stateful_model:predict", data=request_body)
```

Comment:
it might be useful to include in the code snippets also the data for prediction

Reply:
I wanted to make those snippets as small as possible and contain only additional, specific stuff required by stateful models. 
At the end of this section there's a link to `rest_stateful_client.py` example that demonstrates the whole flow including reading data from ARK files, and preparing whole request for every variant (sequence start, sequence end and all the middle requests).